Source/WebCore/editing/EditorCommand.cpp

```diff
@@ -408,9 +408,12 @@ static bool executeForeColor(Frame& frame, Event*, EditorCommandSource source, c
 
 static bool executeFormatBlock(Frame& frame, Event*, EditorCommandSource, const String& value)
 {
-    String tagName = value.convertToASCIILowercase();
-    if (tagName[0] == '<' && tagName[tagName.length() - 1] == '>')
-        tagName = tagName.substring(1, tagName.length() - 2);
+    String lowercaseValue = value.convertToASCIILowercase();
+    AtomString tagName;
+    if (lowercaseValue[0] == '<' && lowercaseValue[lowercaseValue.length() - 1] == '>')
+        tagName = StringView(lowercaseValue).substring(1, lowercaseValue.length() - 2).toAtomString();
+    else
+        tagName = AtomString { lowercaseValue };
```

Comment:
Should this be `WTFMove(lowercaseValue)`? And if that case is not already optimized, it should be, to save a tiny bit of reference count churn.

Separately, I will note that we could fuse the “converting to ASCII lowercase” and “making the atom string” into one operation so we do not need to allocate a temporary string for the case where there’s already an atom string, but the value we have includes an uppercase ASCII character. If that case is common enough then I suppose it would be worth it.